File: src/main/java/com/murex/tbw/domain/book/Book.java
/*******************************************************************************
 *
 * Copyright (c) {2003-2019} Murex S.A.S. and its affiliates.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 *******************************************************************************/

package com.murex.tbw.domain.book;

import com.murex.tbw.domain.country.Language;

public interface Book {

    String getName();

    double getPrice();

    Author getAuthor();

    Language getLanguage();
}

File: kotlin/src/main/kotlin/com/murex/tbw/domain/book/Book.kt
/*******************************************************************************
 *
 * Copyright (c) {2003-2021} Murex S.A.S. and its affiliates.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the MIT License
 * which accompanies this distribution, and is available at
 * https://opensource.org/licenses/MIT
 *
 *******************************************************************************/

package com.murex.tbw.domain.book

import com.murex.tbw.domain.country.Language

interface Book {
    val name: String
    val price: Double
    val author: Author
    val language: Language
}